client/modules/workspace/src/AppsWizard/FormField.tsx

```diff
@@ -9,6 +9,129 @@ import {
 } from '../AppsWizard/AppsFormSchema';
 import { SecondaryButton } from '@client/common-components';
 import { SelectModal } from '../SelectModal/SelectModal';
+import { useSystemOverview, getSystemQueue } from '@client/hooks';
+import { systemStatusStyles, queueStyles } from '@client/workspace';
+
+export type TTapisSystemQueue = {
+  name: string;
+  down: boolean;
+  hidden: boolean;
+  load: number;
+  free: number;
+  running: number;
+  waiting: number;
+};
+
+const ExtendedSelect: React.FC<{
+  after?: React.FC<any>;
+  name: string;
+  value: any;
+  style?: React.CSSProperties;
+  [key: string]: any;
+}> = ({ after: After, name, value, style = {}, ...props }) => {
+  return (
+    <div style={{ display: 'flex', alignItems: 'center' }}>
+      <Select
+        {...props}
+        value={value}
+        style={{ textAlign: 'left', maxWidth: 150, width: '20%', ...style }}
+      />
+      {After && <After name={name} value={value} />}
+    </div>
+  );
+};
+
+const SystemStatus: React.FC<{
+  value: string;
+}> = ({ value }) => {
+  const { getValues } = useFormContext();
+  const { data: systems } = useSystemOverview();
+
+  const getDisplayName = (systemId: string) => {
+    if (!systemId) return '';
+    if (systemId.toLowerCase() === 'ls6') {
+      return 'Lonestar6';
+    }
+    return systemId.charAt(0).toUpperCase() + systemId.slice(1);
+  };
+
+  const displayName = getDisplayName(value);
+  const selectedSystem = systems?.find(
+    (sys) => sys.display_name === displayName
+  );
+
+  return (
+    <div style={{ display: 'flex', alignItems: 'center', marginLeft: 12 }}>
+      <span style={{ marginRight: -5, marginLeft: 8 }}>
+        {value.charAt(0).toUpperCase() + value.slice(1)} status:
+      </span>
+      <div
+        className={`${systemStatusStyles.statusBadge} ${
+          selectedSystem?.is_operational
+            ? systemStatusStyles.open
+            : systemStatusStyles.closed
+        }`}
+        style={{ marginLeft: 12 }}
+      >
+        {selectedSystem?.is_operational ? 'Operational' : 'Maintenance'}
+      </div>
+    </div>
+  );
+};
+
+const QueueStatus: React.FC<{
+  value: string;
+}> = ({ value }) => {
+  const { getValues } = useFormContext();
+  const selectedSystemId = getValues('configuration.execSystemId');
+  const [queueData, setQueueData] = useState<TTapisSystemQueue[]>([]);
```

Comment:
Once we turn `getSystemQueue` into a hook, we can use the data store from the hook instead of this local state.